back/libs/oidc-client/src/dto/identity-service.dto.ts

```diff
@@ -1,14 +1,28 @@
+import { Type } from 'class-transformer';
 import {
   IsArray,
   IsAscii,
   IsJWT,
+  IsObject,
   IsOptional,
   IsString,
   MinLength,
+  ValidateNested,
 } from 'class-validator';
 
 import { TokenResults } from '../interfaces';
 
+export class TokenResultClaimsDto {
```

Comment:
suggestion: `claims` is actually of type `IdTokenClaims` and includes more information than just `amr` and `acr`.
We should find a way to inherit the properties of the `IdTokenClaims` interface from `oidc-provider`.

Another option would be to use the `whitelist` option when validating the `tokenResult`, so the `claims` attribute would be stripped of any additional properties.

```ts
    const tokenValidationErrors = await validate(tokenResult as object, { whitelist: true });
```